Shared/Samples/Show device location history/ShowDeviceLocationHistoryView.swift

```diff
@@ -0,0 +1,234 @@
+        didSet { isShowingAlert = error != nil }
+    }
+    
+    /// A Boolean value indicating whether the tracking button is disabled.
+    @State private var trackingButtonIsDisabled = true
+    
+    /// The view model for this sample.
+    @StateObject private var model = Model()
+    
+    var body: some View {
+        MapView(map: model.map, graphicsOverlays: [model.trackOverlay, model.locationsOverlay])
+            .locationDisplay(model.locationDisplay)
+            .task {
+                do {
+                    try await model.startLocationDataSource()
+                    trackingButtonIsDisabled = false
+                } catch {
+                    // Shows an alert with an error if starting the data source fails.
+                    self.error = error
+                }
+            }
+            .onDisappear {
+                model.stopProcessingLocationChanges()
+                model.isTracking = false
+                model.stopLocationDataSource()
+            }
+            .toolbar {
+                ToolbarItem(placement: .bottomBar) {
+                    Button(model.isTracking ? "Stop Tracking" : "Start Tracking") {
+                        model.isTracking.toggle()
+                    }
+                    .disabled(trackingButtonIsDisabled)
+                }
+            }
+            .alert(isPresented: $isShowingAlert, presentingError: error)
+    }
+}
+
+private extension ShowDeviceLocationHistoryView {
+    /// The model used to store the geo model and other expensive objects
+    /// used in this view.
+    @MainActor
+    class Model: ObservableObject {
+        /// A Boolean value indicating whether to track the device location.
+        @Published var isTracking = false {
+            didSet {
+                if isTracking {
+                    startProcessingLocationChanges()
+                } else {
+                    stopProcessingLocationChanges()
+                }
+            }
+        }
+        
+        /// A map with a light gray basemap style.
+        let map = Map(basemapStyle: .arcGISLightGrayBase)
+        
+        /// A location display using a simulated location data source.
+        let locationDisplay: LocationDisplay
+        
+        /// A graphics overlay for location history polyline.
+        let trackOverlay: GraphicsOverlay
+        
+        /// A graphics overlay for location history points.
+        let locationsOverlay: GraphicsOverlay
+        
+        /// The markers for traveled location points.
+        private let locationGraphic = Graphic()
+        
+        /// The line graphic for traveled path.
+        private let trackGraphic = Graphic()
+        
+        /// A polyline builder to make the line graphic.
+        private let trackBuilder = PolylineBuilder(spatialReference: .webMercator)
+        
+        /// A multipoint builder to make the point graphics.
+        private let pointBuilder = MultipointBuilder(spatialReference: .webMercator)
```

Comment:
This name is confusing because there is a type called `PointBuilder`. A better name would be `multipointBuilder`.